### PR TITLE
Rework post connections

### PR DESCRIPTION
**Pull request by liach**
**The Issue**
Improve post connections, metal cube beacon base, gated track connections, try add post washing
 
**The Proposal**
This is a try to clean up post connections so that we don't need to mingle with it later.
 
**Possible Side Effects**
Changed signal connections to single thick.
Some other visual changes and connection changes as well.
Will work on adding concrete walls later.
 
**Alternatives**
Still a wip, posting for cj to review.

### CODE REVIEW

**Comment by CovertJaguar** [comment]
> Changed signal connections to single thick.

Why?

File: src/main/java/mods/railcraft/common/blocks/aesthetics/metals/EnumMetal.java [hunk 1]
```diff
@@ -23,16 +23,17 @@
...
-    BLOCK_TIN(ModuleResources.class, "tin", new SimpleCube(), 3f, 10f),
-    BLOCK_LEAD(ModuleResources.class, "lead", new SimpleCube(), 2f, 20f),
-    BLOCK_STEEL(ModuleResources.class, "steel", new SimpleCube(), 5f, 15f),
-    BLOCK_SILVER(ModuleResources.class, "silver", new SimpleCube(), 3f, 10f),
-    BLOCK_BRONZE(ModuleResources.class, "bronze", new SimpleCube(), 3f, 10f),
-    BLOCK_NICKEL(ModuleResources.class, "nickel", new SimpleCube(), 3f, 10f),
-    BLOCK_INVAR(ModuleResources.class, "invar", new SimpleCube(), 3f, 10f),
-    BLOCK_ZINC(ModuleResources.class, "zinc", new SimpleCube(), 3f, 10f),
-    BLOCK_BRASS(ModuleResources.class, "brass", new SimpleCube(), 3f, 10f),;
+    BLOCK_COPPER(ModuleResources.class, "copper", new MetalCube(), 3f, 10f),
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/aesthetics/metals/EnumMetal.java:26]: Is there any reason not to just strip out the SimpleCube def? Its a hold over from BlockGeneric.

File: src/main/java/mods/railcraft/common/blocks/aesthetics/post/BlockPostBase.java [hunk 2]
```diff
@@ -67,15 +69,15 @@ protected BlockPostBase() {
...
             return Column.FULL;
         BlockPos up = pos.up();
         IBlockState above = WorldPlugin.getBlockState(world, up);
-        if (above instanceof BlockPostBase)
+        if (above.getBlock() instanceof BlockPostBase)
             return Column.FULL;
         if (!isPlatform(state) && !WorldPlugin.isBlockAir(world, up, above))
             return Column.MINI;
-        return Column.NONE;
+        return Column.FULL;
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/aesthetics/post/BlockPostBase.java:80]: Doesn't this result in crossbeams having a pillar?

File: src/main/java/mods/railcraft/common/blocks/aesthetics/post/PostConnectionHelper.java [hunk 3]
```diff
@@ -11,72 +11,24 @@
...
+public final class PostConnectionHelper {
 
-    public static ConnectStyle connect(IBlockAccess world, BlockPos pos, IBlockState state, EnumFacing side) {
-        try {
-            Block block = state.getBlock();
-            if (block instanceof IPostConnection && ((IPostConnection) block).connectsToPost(world, pos, state, side) == ConnectStyle.NONE)
-                return ConnectStyle.NONE;
-        } catch (Error error) {
-            Game.log().api(Railcraft.NAME, error, IPostConnection.class);
+    public static ConnectStyle connectMetalPost(IBlockAccess world, BlockPos pos, IBlockState state, EnumFacing side) {
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/aesthetics/post/PostConnectionHelper.java:28]: Why rename it? It affects all posts.

File: src/main/java/mods/railcraft/common/blocks/machine/wayobjects/signals/BlockMachineSignal.java [hunk 4]
```diff
@@ -107,27 +110,35 @@ public IBlockState getActualState(IBlockState state, IBlockAccess worldIn, Block
...
+        IBlockState state = WorldPlugin.getBlockState(world, fromPos.offset(fromSide));
+        Block block = state.getBlock();
+        return block instanceof BlockPostBase;
+    }
+
     @Override
     public ConnectStyle connectsToPost(IBlockAccess world, BlockPos pos, IBlockState state, EnumFacing face) {
-        return ConnectStyle.TWO_THIN;
+        EnumFacing front = WorldPlugin.getTileEntity(world, pos, TileSignalBase.class).map(TileSignalBase::getFacing).orElse(EnumFacing.NORTH);
+        return face != front ? ConnectStyle.SINGLE_THICK : ConnectStyle.NONE;
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/blocks/machine/wayobjects/signals/BlockMachineSignal.java:141]: Pretty sure SINGLE_THINK will look horrible suspended above the tracks.

**Review by CovertJaguar — CHANGES_REQUESTED** [review]
Nope:


Um....




Can you tell the difference between these two pictures?
Before:

After:

File: src/main/java/mods/railcraft/common/blocks/aesthetics/post/PostConnectionHelper.java (hunk 3 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/aesthetics/post/PostConnectionHelper.java:28]: Don't think we use this for concrete posts.

File: src/main/java/mods/railcraft/common/blocks/machine/wayobjects/signals/BlockMachineSignal.java (hunk 4 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/machine/wayobjects/signals/BlockMachineSignal.java:141]: Hmm, when was single thick used for initially?

File: src/main/java/mods/railcraft/common/blocks/aesthetics/post/BlockPostBase.java (hunk 2 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/aesthetics/post/BlockPostBase.java:80]: Hmm, that if above should handle it. This is to fix the no model thing when the post is in air

File: src/main/java/mods/railcraft/common/blocks/aesthetics/metals/EnumMetal.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/blocks/aesthetics/metals/EnumMetal.java:26]: Thought it was able to hold and share logic. Will change

File: src/main/java/mods/railcraft/common/blocks/aesthetics/post/PostConnectionHelper.java (hunk 3 above)
Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/blocks/aesthetics/post/PostConnectionHelper.java:28]: We did.

File: src/main/java/mods/railcraft/common/blocks/machine/wayobjects/signals/BlockMachineSignal.java (hunk 4 above)
Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/blocks/machine/wayobjects/signals/BlockMachineSignal.java:141]: Mounting signs and lanterns on posts.

File: src/main/java/mods/railcraft/common/blocks/aesthetics/post/BlockPostBase.java (hunk 2 above)
Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/blocks/aesthetics/post/BlockPostBase.java:80]: It definitely adds vertical beams to suspended posts, see my screen shots.

**Pull request closed without merging**